python-client/tests/pytorch/test_NewspaperClassification_PyTorchDataset.py

```diff
@@ -0,0 +1,255 @@
+import pandas as pd
+import numpy as np
+from scipy import special
+import pytest
+import time
+import torch
+from torchtext.datasets import AG_NEWS
+train_iter = iter(AG_NEWS(split='train'))
+
+from torchtext.data.utils import get_tokenizer
+from torchtext.vocab import build_vocab_from_iterator
+from torch.utils.data.dataset import random_split
+from torchtext.data.functional import to_map_style_dataset
+from torch.utils.data import DataLoader
+from torch.utils.data import Dataset as torch_dataset
+
+from torch import nn
+
+from giskard.core.model_validation import validate_model
+from giskard.client.giskard_client import GiskardClient
+from giskard import PyTorchModel, Dataset
+
+import re
+import httpretty
+url = "http://giskard-host:12345"
+token = "SECRET_TOKEN"
+auth = "Bearer SECRET_TOKEN"
+content_type = "application/json"
+model_name = "uploaded model"
+b_content_type = b"application/json"
+
+class TextClassificationModel(nn.Module):
+
+    def __init__(self, vocab_size, embed_dim, num_class):
+        super(TextClassificationModel, self).__init__()
+        self.embedding = nn.EmbeddingBag(vocab_size, embed_dim, sparse=True)
+        self.fc = nn.Linear(embed_dim, num_class)
+        self.init_weights()
+
+    def init_weights(self):
+        initrange = 0.5
+        self.embedding.weight.data.uniform_(-initrange, initrange)
+        self.fc.weight.data.uniform_(-initrange, initrange)
+        self.fc.bias.data.zero_()
+
+    def forward(self, text, offsets):
+        embedded = self.embedding(text, offsets)
+        return self.fc(embedded)
+
+#@pytest.mark.skip(reason="WIP")
+@httpretty.activate(verbose=True, allow_net_connect=False)
+def test_text_sentiment_ngrams_tutorial():
+
+    train_iter = AG_NEWS(split='train')
+    num_class = len(set([label for (label, text) in train_iter]))
+
+    tokenizer = get_tokenizer('basic_english')
+
+    def yield_tokens(data_iter):
+        for _, text in data_iter:
+            yield tokenizer(text)
+
+    train_iter = AG_NEWS(split='train')
```

Comment:
train_iter doesn't get exhausted after one iteration, you can re-use the same object